daemon/mgr/container_validation.go

```diff
@@ -188,3 +196,77 @@ func (mgr *ContainerManager) validateLogConfig(c *Container) error {
 		return fmt.Errorf("not support (%v) log driver yet", logCfg.LogDriver)
 	}
 }
+
+// validateNvidiaConfig
+func validateNvidiaConfig(r *types.Resources) error {
+	if r.NvidiaConfig == nil {
+		return nil
+	}
+
+	if err := validateNvidiaDriver(r); err != nil {
+		return err
+	}
+
+	if err := validateNvidiaDevice(r); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateNvidiaDriver(r *types.Resources) error {
+	n := r.NvidiaConfig
+	n.NvidiaDriverCapabilities = strings.TrimSpace(n.NvidiaDriverCapabilities)
+	if n.NvidiaDriverCapabilities == "" {
+		// use default driver capability: utility
+		return nil
+	}
+
+	if n.NvidiaDriverCapabilities == "all" {
+		// enable all capabilities
+		return nil
+	}
+
+	supportedDrivers := []string{"compute", "compat32", "graphics", "utility", "video", "display"}
```

Comment:
could we use map global var to hold the supportedDrivers, like

```
var supportedNvidiaDrivers = map[string]struct{}{
        "compute": struct{},
        ....
}
```

therefore, we don't need to allocate new memory in the `utils.StringInSlice`.